### DISCUSSION TITLE
Dynamic root cannot start at root content?

### DISCUSSION THREAD

**Discussion by james-whittington1** [umbraco/Umbraco-CMS#15704] [Features and ideas]
I have a site where a settings node is outside of the site root, and there is no way to select it using the new dynamic root except picking it by key. 

As an interim solution I have built a query step that gets the pages at root, but it seems like this is an origin, as it step it just discards whatever else precedes it. 

An origin of "Root content" which returns the same as  _contentService.GetRootContent() would be a reasonably useful kind for websites that have multiple root content nodes  I think?

**Comment by bergmania** [comment]
Sounds like a okay idea, but for the specific use case, what are the problems with choosing the settings node by key? The key id not the Id, so it will be the same on all environments.

**Reply by james-whittington1** [reply]
Hello, sorry I got really busy and didn't have time to reply. I've just come up against this same issue again and I don't understand what you mean by picking by key? 

all my page content is under home, and then Home's sibling "settings" holds shared data that can be used in pods etc.

![image](https://github.com/umbraco/Umbraco-CMS/assets/37091895/66f27ce0-9469-43f5-9a65-cb04c60ccaea)

I would like a dynamic root item that allows me to start the multinode tree picker at the settings/health funds root, where the user would be able to see "Fund 1", and "Fund 2" 

previously we just put xpath //settings and that was ok, but not it has started to show errors, and we want to make sure we're not making any potential upgrades to v14 more difficult. 

Site
+ Home
++ all pages
+ Settings
++ Health funds
+++ fund 1
+++ fund 2

**Reply by james-whittington1** [reply]
We are using umbraco deploy so the key will (or could) change between environments as far as I am aware

**Reply by skttl** [reply]
> We are using umbraco deploy so the key will (or could) change between environments as far as I am aware

Keys won't, id's will.

**Reply by james-whittington1** [reply]
Incredible. That is handy. 

I do think it would be useful to have the option of selecting dynamic roots @ level 1. 

For example, we have sites that are not using deploy and the key will change (if the content is manually authored in different environments). 

We are using usync (for settings only) in these situations so the configuration should be the the same for each environment, so the ability to do things based on node structure rather than the key is still really important.

**Reply by simonech** [reply]
> > We are using umbraco deploy so the key will (or could) change between environments as far as I am aware
> 
> Keys won't, id's will.

Not everyone uses Umbraco Deploy

**Reply by simonech** [reply]
> The key id not the Id, so it will be the same on all environments.

only if you sync all content via Umbraco deploy or usync publisher. Many ppl don't use either, and even if they did, normally you don't do that between multiple developers.

**Comment by james-whittington1** [comment]
Hi again, with the xpath method now being removed in 14, I think this is still relevant. 

I have some sites that that I look after that are not on cloud, and just push schema changes between environments using usync. Since the content is constructed in parallel they keys are not stable. 

If there was a dynamic root that could start above the content noes this would solve all the problems


-1, type:settingsNode -> use dynamic root with zero issues from here.

-1, type:homepageNode -> use dynamic root with zero issues from here too. 


Dynamic root is a really good solution but I feel like I am banging my head against a wall with the way I am using it. Am I just doing it wrong, or should I expect that setting up my site with some root nodes being used to hold shared content should work gracefully.

**Comment by simonech** [comment]
I agree, I just started migrating our xpath pickers to dynamic roots, and indeed we use a lot references across roots.
Now we are building a lot of custom query step to fetch picker start nodes under other roots.
But if there was a top level root start item equivalent to `_umbracoContextFactory.EnsureUmbracoContext().UmbracoContext.Content?.GetAtRoot()` it would save a lot of custom code.

And the comment of umbraco deploy is not relevant for us, since we do not use it.

**Comment by james-whittington1** [comment]
Quite disappointed with the lack of any clear resolution on this one. I don't mind if the official answer is "Don't do this" but just dropping "use Umbraco deploy" and dipping out is fairly unhelpful.

**Comment by AndyButland** [comment]
A "Content Root" origin step has been added in https://github.com/umbraco/Umbraco-CMS/pull/21161 and scheduled for 17.2.  I believe that resolves this discussion, so will close it off now.

**Comment by james-whittington1** [comment]
This is great news, thanks for the update.
